velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -60,20 +61,148 @@ void WindowFuzzer::addWindowFunctionSignatures(
   }
 }
 
-std::tuple<std::string, bool> WindowFuzzer::generateFrameClause() {
-  auto frameType = [](int value) -> const std::string {
-    switch (value) {
-      case 0:
-        return "RANGE";
-      case 1:
-        return "ROWS";
-      default:
-        VELOX_UNREACHABLE("Unknown value for frame type generation");
+std::tuple<
+    core::WindowNode::WindowType,
+    core::WindowNode::BoundType,
+    core::WindowNode::BoundType>
+WindowFuzzer::frameWindowTypeAndBoundType() {
+  // Randomly select if ROWS or RANGE frame
+  auto windowType = vectorFuzzer_.coinToss(0.1)
+      ? core::WindowNode::WindowType::kRows
+      : core::WindowNode::WindowType::kRange;
+
+  std::vector<core::WindowNode::BoundType> startBoundOptions, endBoundOptions;
+  startBoundOptions = {
+      core::WindowNode::BoundType::kUnboundedPreceding,
+      core::WindowNode::BoundType::kPreceding,
+      core::WindowNode::BoundType::kCurrentRow,
+      core::WindowNode::BoundType::kFollowing};
+  endBoundOptions = {
+      core::WindowNode::BoundType::kPreceding,
+      core::WindowNode::BoundType::kCurrentRow,
+      core::WindowNode::BoundType::kFollowing,
+      core::WindowNode::BoundType::kUnboundedFollowing};
+
+  // End bound option should not be greater than start bound option as this
```

Comment:
This comment is confusing because endBoundOption is always greater than start bound options.